gdal: switch to openexr_3

autra opened the pull request:
There are several CVEs affecting openexr v2 and it seems to disturb people. As gdal supports openexr v3, let's live in the future and upgrade this dep.

Fix #366916


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

autra committed:
gdal: switch to openexr_3

There are several CVEs affecting openexr v2 and it seems to disturb
people. As gdal supports openexr v3, let's live in the future and
upgrade this dep.

Fix #366916

imincik commented:
I agree with this change. Thanks for acting on this so quickly @autra .

LeSuisse commented:
Flagging as security related.

imincik commented:
Result of `nixpkgs-review pr 366939` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>python312Packages.python-mapnik</li>
    <li>python312Packages.python-mapnik.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.worldengine</li>
    <li>python312Packages.worldengine.dist</li>
    <li>t-rex</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>apacheHttpdPackages.mod_tile</li>
  </ul>
</details>
<details>
  <summary>162 packages built:</summary>
  <ul>
    <li>cloudcompare</li>
    <li>entwine</li>
    <li>gdal (python312Packages.gdal)</li>
    <li>gdalMinimal</li>
    <li>gmt</li>
    <li>gplates</li>
    <li>grass</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>mysql-workbench</li>
    <li>octavePackages.mapping</li>
    <li>openorienteering-mapper</li>
    <li>otb</li>
    <li>paraview</li>
    <li>pdal</li>
    <li>perl538Packages.Tirex</li>
    <li>perl538Packages.Tirex.devdoc</li>
    <li>perl540Packages.Tirex</li>
    <li>perl540Packages.Tirex.devdoc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresql14JitPackages.postgis</li>
    <li>postgresql14JitPackages.postgis.doc</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>postgresql15JitPackages.postgis</li>
    <li>postgresql15JitPackages.postgis.doc</li>
    <li>postgresql15Packages.postgis</li>
    <li>postgresql15Packages.postgis.doc</li>
    <li>postgresql16JitPackages.postgis</li>
    <li>postgresql16JitPackages.postgis.doc</li>
    <li>postgresql16Packages.postgis</li>
    <li>postgresql16Packages.postgis.doc</li>
    <li>postgresql17JitPackages.postgis</li>
    <li>postgresql17JitPackages.postgis.doc</li>
    <li>postgresql17Packages.postgis</li>
    <li>postgresql17Packages.postgis.doc</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.django-bootstrap4</li>
    <li>python311Packages.django-bootstrap4.dist</li>
    <li>python311Packages.django-bootstrap5</li>
    <li>python311Packages.django-bootstrap5.dist</li>
    <li>python311Packages.drf-extra-fields</li>
    <li>python311Packages.drf-extra-fields.dist</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geoarrow-pandas</li>
    <li>python311Packages.geoarrow-pandas.dist</li>
    <li>python311Packages.geoarrow-pyarrow</li>
    <li>python311Packages.geoarrow-pyarrow.dist</li>
    <li>python311Packages.geodatasets</li>
    <li>python311Packages.geodatasets.dist</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.geoparquet</li>
    <li>python311Packages.geoparquet.dist</li>
    <li>python311Packages.inequality</li>
    <li>python311Packages.inequality.dist</li>
    <li>python311Packages.libpysal</li>
    <li>python311Packages.libpysal.dist</li>
    <li>python311Packages.mapclassify</li>
    <li>python311Packages.mapclassify.dist</li>
    <li>python311Packages.momepy</li>
    <li>python311Packages.momepy.dist</li>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.msticpy</li>
    <li>python311Packages.msticpy.dist</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-documents.dist</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pyogrio</li>
    <li>python311Packages.pyogrio.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.rioxarray</li>
    <li>python311Packages.rioxarray.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>python312Packages.bsuite</li>
    <li>python312Packages.bsuite.dist</li>
    <li>python312Packages.cartopy</li>
    <li>python312Packages.cartopy.dist</li>
    <li>python312Packages.django-bootstrap4</li>
    <li>python312Packages.django-bootstrap4.dist</li>
    <li>python312Packages.django-bootstrap5</li>
    <li>python312Packages.django-bootstrap5.dist</li>
    <li>python312Packages.drf-extra-fields</li>
    <li>python312Packages.drf-extra-fields.dist</li>
    <li>python312Packages.fiona</li>
    <li>python312Packages.fiona.dist</li>
    <li>python312Packages.folium</li>
    <li>python312Packages.folium.dist</li>
    <li>python312Packages.geoarrow-pandas</li>
    <li>python312Packages.geoarrow-pandas.dist</li>
    <li>python312Packages.geoarrow-pyarrow</li>
    <li>python312Packages.geoarrow-pyarrow.dist</li>
    <li>python312Packages.geodatasets</li>
    <li>python312Packages.geodatasets.dist</li>
    <li>python312Packages.geopandas</li>
    <li>python312Packages.geopandas.dist</li>
    <li>python312Packages.geoparquet</li>
    <li>python312Packages.geoparquet.dist</li>
    <li>python312Packages.inequality</li>
    <li>python312Packages.inequality.dist</li>
    <li>python312Packages.libpysal</li>
    <li>python312Packages.libpysal.dist</li>
    <li>python312Packages.mapclassify</li>
    <li>python312Packages.mapclassify.dist</li>
    <li>python312Packages.momepy</li>
    <li>python312Packages.momepy.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.msticpy</li>
    <li>python312Packages.msticpy.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.osmnx</li>
    <li>python312Packages.osmnx.dist</li>
    <li>python312Packages.plotnine</li>
    <li>python312Packages.plotnine.dist</li>
    <li>python312Packages.pygmt</li>
    <li>python312Packages.pygmt.dist</li>
    <li>python312Packages.pyogrio</li>
    <li>python312Packages.pyogrio.dist</li>
    <li>python312Packages.rasterio</li>
    <li>python312Packages.rasterio.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
    <li>python312Packages.wktutils</li>
    <li>python312Packages.wktutils.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>saga</li>
    <li>sumo</li>
    <li>survex</li>
    <li>therion</li>
    <li>tunnelx</li>
    <li>udig</li>
    <li>vpv</li>
  </ul>
</details>

imincik reviewed (approved):
Looks good to me. nixpkgs-review is passing .

imincik commented:
I think, even more correct solution would be to update [default openexr version from 2 to 3](https://github.com/NixOS/nixpkgs/blob/296c5eb9769d5acf7eb18118365798b9c05b286c/pkgs/top-level/all-packages.nix#L10368)

autra commented:
> I think, even more correct solution would be to update [default openexr version from 2 to 3](https://github.com/NixOS/nixpkgs/blob/296c5eb9769d5acf7eb18118365798b9c05b286c/pkgs/top-level/all-packages.nix#L10368) . But that's much bigger task.

Indeed. Should v2 be marked as insecure? (can we affort that?) It's a bit above my paygrade but [some of these CVEs](https://www.cvedetails.com/vulnerability-list/vendor_id-9960/product_id-17799/version_id-1763700/Openexr-Openexr-2.5.10.html) seems quite serious. Someone more knowledgeable in security should check though :-)

autra commented:
I've opened a followup: https://github.com/NixOS/nixpkgs/pull/367406  for the rest of the packages.

imincik commented:
> > I think, even more correct solution would be to update [default openexr version from 2 to 3](https://github.com/NixOS/nixpkgs/blob/296c5eb9769d5acf7eb18118365798b9c05b286c/pkgs/top-level/all-packages.nix#L10368) . But that's much bigger task.
> 
> Indeed. Should v2 be marked as insecure? (can we affort that?) It's a bit above my paygrade but [some of these CVEs](https://www.cvedetails.com/vulnerability-list/vendor_id-9960/product_id-17799/version_id-1763700/Openexr-Openexr-2.5.10.html) seems quite serious. Someone more knowledgeable in security should check though :-)

Make a sense to mark v2 as insecure.

LeSuisse commented:
> > I think, even more correct solution would be to update [default openexr version from 2 to 3](https://github.com/NixOS/nixpkgs/blob/296c5eb9769d5acf7eb18118365798b9c05b286c/pkgs/top-level/all-packages.nix#L10368) . But that's much bigger task.
> 
> Indeed. Should v2 be marked as insecure? (can we affort that?) It's a bit above my paygrade but [some of these CVEs](https://www.cvedetails.com/vulnerability-list/vendor_id-9960/product_id-17799/version_id-1763700/Openexr-Openexr-2.5.10.html) seems quite serious. Someone more knowledgeable in security should check though :-)

Before removal it would probably make sense to move all the packages currently using `openexr` 2 to `openexr_3`, then mark `openexr` 2 with `knownVulnerabilities`. This should allow us to backport the changes to stable without breakages (hopefully...). Then if we have no more dependents we can remove `openexr_2` completely.
Note that moving some of the packages to `openexr_3` will need to go through `staging` (e.g. `gst_all_1.gst-plugins-bad`)

autra commented:
> Note that moving some of the packages to openexr_3 will need to go through staging (e.g. gst_all_1.gst-plugins-bad)

And actually I guess this PR should go to staging too.

autra commented:
I  changed the base to staging, and I'm going to do that as well for https://github.com/NixOS/nixpkgs/pull/367406

imincik commented:
> > Note that moving some of the packages to openexr_3 will need to go through staging (e.g. gst_all_1.gst-plugins-bad)
> 
> And actually I guess this PR should go to staging too.

I don't think so.

Number of affected packages in this PR is following:
```
Package: added 0, removed 0, changed 97, Rebuild: linux 97, darwin 6
```

autra commented:
> > > Note that moving some of the packages to openexr_3 will need to go through staging (e.g. gst_all_1.gst-plugins-bad)
> > 
> > 
> > And actually I guess this PR should go to staging too.
> 
> I don't think so.
> 
> Number of affected packages in this PR is following:
> 
> ```
> Package: added 0, removed 0, changed 97, Rebuild: linux 97, darwin 6
> ```

The labels are off then? When I tried nixpkgs-review, I had a higher amounts of rebuids... I'll test again a bit later

imincik commented:
We don't need to go via `staging` branch for gdal PRs. My `nixpkgs-review` built 162 packages. Limit for staging is > 500.

autra commented:
Ok, fair enough, I rebased and switched the base branch back to master.

autra commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 366939`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 13 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.python-mapnik</li>
    <li>python312Packages.python-mapnik.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.worldengine</li>
    <li>python312Packages.worldengine.dist</li>
    <li>python313Packages.python-mapnik</li>
    <li>python313Packages.python-mapnik.dist</li>
    <li>python313Packages.worldengine</li>
    <li>python313Packages.worldengine.dist</li>
    <li>t-rex</li>
  </ul>
</details>
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>cloudcompare</li>
    <li>otb</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>python312Packages.django-bootstrap4</li>
    <li>python312Packages.django-bootstrap4.dist</li>
    <li>python312Packages.django-bootstrap5</li>
    <li>python312Packages.django-bootstrap5.dist</li>
    <li>python312Packages.drf-extra-fields</li>
    <li>python312Packages.drf-extra-fields.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python313Packages.drf-extra-fields</li>
    <li>python313Packages.drf-extra-fields.dist</li>
    <li>python313Packages.netbox-documents</li>
    <li>python313Packages.netbox-documents.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 105 packages built:</summary>
  <ul>
    <li>apacheHttpdPackages.mod_tile</li>
    <li>entwine</li>
    <li>gdal (python312Packages.gdal)</li>
    <li>gdalMinimal</li>
    <li>gmt</li>
    <li>gplates</li>
    <li>grass</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>mysql-workbench</li>
    <li>octavePackages.mapping</li>
    <li>openorienteering-mapper</li>
    <li>paraview</li>
    <li>pdal</li>
    <li>perl538Packages.Tirex</li>
    <li>perl538Packages.Tirex.devdoc</li>
    <li>perl540Packages.Tirex</li>
    <li>perl540Packages.Tirex.devdoc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresql14JitPackages.postgis</li>
    <li>postgresql14JitPackages.postgis.doc</li>
    <li>postgresql15JitPackages.postgis</li>
    <li>postgresql15JitPackages.postgis.doc</li>
    <li>postgresql15Packages.postgis</li>
    <li>postgresql15Packages.postgis.doc</li>
    <li>postgresql16JitPackages.postgis</li>
    <li>postgresql16JitPackages.postgis.doc</li>
    <li>postgresql16Packages.postgis</li>
    <li>postgresql16Packages.postgis.doc</li>
    <li>postgresql17JitPackages.postgis</li>
    <li>postgresql17JitPackages.postgis.doc</li>
    <li>postgresql17Packages.postgis</li>
    <li>postgresql17Packages.postgis.doc</li>
    <li>python312Packages.bsuite</li>
    <li>python312Packages.bsuite.dist</li>
    <li>python312Packages.cartopy</li>
    <li>python312Packages.cartopy.dist</li>
    <li>python312Packages.fiona</li>
    <li>python312Packages.fiona.dist</li>
    <li>python312Packages.folium</li>
    <li>python312Packages.folium.dist</li>
    <li>python312Packages.geoarrow-pandas</li>
    <li>python312Packages.geoarrow-pandas.dist</li>
    <li>python312Packages.geoarrow-pyarrow</li>
    <li>python312Packages.geoarrow-pyarrow.dist</li>
    <li>python312Packages.geodatasets</li>
    <li>python312Packages.geodatasets.dist</li>
    <li>python312Packages.geopandas</li>
    <li>python312Packages.geopandas.dist</li>
    <li>python312Packages.geoparquet</li>
    <li>python312Packages.geoparquet.dist</li>
    <li>python312Packages.inequality</li>
    <li>python312Packages.inequality.dist</li>
    <li>python312Packages.libpysal</li>
    <li>python312Packages.libpysal.dist</li>
    <li>python312Packages.mapclassify</li>
    <li>python312Packages.mapclassify.dist</li>
    <li>python312Packages.momepy</li>
    <li>python312Packages.momepy.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.msticpy</li>
    <li>python312Packages.msticpy.dist</li>
    <li>python312Packages.osmnx</li>
    <li>python312Packages.osmnx.dist</li>
    <li>python312Packages.plotnine</li>
    <li>python312Packages.plotnine.dist</li>
    <li>python312Packages.pygmt</li>
    <li>python312Packages.pygmt.dist</li>
    <li>python312Packages.pyogrio</li>
    <li>python312Packages.pyogrio.dist</li>
    <li>python312Packages.rasterio</li>
    <li>python312Packages.rasterio.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
    <li>python312Packages.wktutils</li>
    <li>python312Packages.wktutils.dist</li>
    <li>python313Packages.django-bootstrap4</li>
    <li>python313Packages.django-bootstrap4.dist</li>
    <li>python313Packages.django-bootstrap5</li>
    <li>python313Packages.django-bootstrap5.dist</li>
    <li>python313Packages.gdal</li>
    <li>python313Packages.pygmt</li>
    <li>python313Packages.pygmt.dist</li>
    <li>python313Packages.pyogrio</li>
    <li>python313Packages.pyogrio.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>saga</li>
    <li>sumo</li>
    <li>survex</li>
    <li>therion</li>
    <li>tunnelx</li>
    <li>udig</li>
    <li>vpv</li>
  </ul>
</details>
